addons/hw_escpos/escpos/escpos.py

```diff
@@ -885,14 +885,24 @@ def cut(self, mode=''):
             self._raw(PAPER_FULL_CUT)
 
 
-    def cashdraw(self, pin):
-        """ Send pulse to kick the cash drawer """
-        if pin == 2:
-            self._raw(CD_KICK_2)
-        elif pin == 5:
-            self._raw(CD_KICK_5)
-        else:
-            raise CashDrawerError()
+    def cashdraw(self, pin, tries=5):
+        """ Send pulse to kick the cash drawer
+
+        With some printers the drawer will not open after one pulse, for this reason we will check
+        the drawer status up to 'tries' times and send a new pulse if the drawer is still closed. If the
+        drawer status is open, we will stop sending pulses.
+        """
+        try_range = tries - 1
+        for i in range(try_range):
+            if pin == 2:
+                self._raw(CD_KICK_2)
+            elif pin == 5:
+                self._raw(CD_KICK_5)
+            else:
+                raise CashDrawerError()
+
+            if i != try_range and not self.get_printer_status()['printer']['drawer_pin_high']:
```

Comment:
I think 'i != try_range' is useless as we are in 'range(try_range)' otherwise it's ok for me

Reply:
It would indeed never reach this value, it had to be `range(tries)` and `i != tries-1` :see_no_evil: 
If it's ok for you you may do robodoo r+ :+1: